[WPE][GTK] generate-pdfjs-gresource-manifest.py should be more careful about unknown files

mcatanzaro opened the pull request:
#### 6a288a67f2d647921fbf68bf582f3a334aaed1f1
<pre>
[WPE][GTK] generate-pdfjs-gresource-manifest.py should be more careful about unknown files
<a href="https://bugs.webkit.org/show_bug.cgi?id=240536">https://bugs.webkit.org/show_bug.cgi?id=240536</a>

Patch by Michael Catanzaro &lt;mcatanzaro@redhat.com &gt; on 2022-05-17
Reviewed by Adrian Perez de Castro.

Instead of silently ignoring unexpected files, let&apos;s create a list of
all files that we do not want to include in the resource manifest.
Anything unexpected will result in a build failure so that there are no
mistakes.

Also, change the lists at the top of the script into sets. Adrian
noticed that they do not need to be ordered.

* Tools/glib/generate-pdfjs-gresource-manifest.py:
(get_filenames):
(get_filenames.should_ignore_resource): Deleted.

Canonical link: <a href="https://commits.webkit.org/250671@main">https://commits.webkit.org/250671@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294379">https://svn.webkit.org/repository/webkit/trunk@294379</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

aperezdc commented on Tools/glib/generate-pdfjs-gresource-manifest.py:
```diff
@@ -23,14 +23,16 @@
+          'README.webkit',
+          'web/cmaps/LICENSE',
+          'web/standard_fonts/LICENSE_FOXIT',
+          'web/standard_fonts/LICENSE_LIBERATION']
```
I would use a set literal here, but the list is small so in this case using a set instead of a list might not be any advantage. So up to you whether you want to change it or not before landing.

```suggestion
IGNORE = {'LICENSE',
          'README.webkit',
          'web/cmaps/LICENSE',
          'web/standard_fonts/LICENSE_FOXIT',
          'web/standard_fonts/LICENSE_LIBERATION'}
```

mcatanzaro replied:
I've changed all of these lists to sets.

aperezdc reviewed (approved):
LGTM

Commit:
[git-webkit] Make git-webkit revert a two step workflow
https://bugs.webkit.org/show_bug.cgi?id=240192
rdar://92875608

Reviewed by Jonathan Bedard.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py:
(PullRequest.create_pull_request):
(PullRequest.add_comment_to_reverted_commit_bug_tracker):
(PullRequest.is_revert_commit):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py:
(Revert):
(Revert.parser):
(Revert.revert_commit):
(Revert.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/revert_unittest.py:
(TestRevert.test_github):
(test_update):

Canonical link: https://commits.webkit.org/250668@main


git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294376 268f45cc-cd09-0410-ab3c-d52691b4dbfc

basuke committed:
[WinCairo] Add memory usage of LibcMalloc category
https://bugs.webkit.org/show_bug.cgi?id=239660

Reviewed by Ross Kirsling.

WinCairo's memory timeline just displays JavaScript memory usage. Adding
this info shows also Page
memory usage.

* page/win/ResourceUsageThreadWin.cpp:
(WebCore::ResourceUsageThread::platformCollectMemoryData):

Canonical link: https://commits.webkit.org/250669@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294377 268f45cc-cd09-0410-ab3c-d52691b4dbfc

aproskuryakov committed:
Reverting https://trac.webkit.org/r294367, as some of our automation hosts don't
have the --date=unix option.

Reviewed by NOBODY (OOPS!).

* Scripts/libraries/webkitscmpy/setup.py:
* Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py:
* Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git.Cache.populate):
(Git.commit):
(Git._args_from_content):
(Git.commits):
* Scripts/libraries/webkitscmpy/webkitscmpy/mocks/local/git.py:
* Scripts/libraries/webkitscmpy/webkitscmpy/test/git_unittest.py:
(TestGit.test_log):

Canonical link: https://commits.webkit.org/250670@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294378 268f45cc-cd09-0410-ab3c-d52691b4dbfc

mcatanzaro committed:
[WPE][GTK] generate-pdfjs-gresource-manifest.py should be more careful about unknown files
https://bugs.webkit.org/show_bug.cgi?id=240536

Patch by Michael Catanzaro <mcatanzaro@redhat.com> on 2022-05-17
Reviewed by Adrian Perez de Castro.

Instead of silently ignoring unexpected files, let's create a list of
all files that we do not want to include in the resource manifest.
Anything unexpected will result in a build failure so that there are no
mistakes.

Also, change the lists at the top of the script into sets. Adrian
noticed that they do not need to be ordered.

* Tools/glib/generate-pdfjs-gresource-manifest.py:
(get_filenames):
(get_filenames.should_ignore_resource): Deleted.

Canonical link: https://commits.webkit.org/250671@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294379 268f45cc-cd09-0410-ab3c-d52691b4dbfc
